src/operator/mkldnn/mkldnn_memory-inl.h

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file mkldnn_memory-inl.h 
+ * \brief
+ * \author Chen, Xiaoming 
+*/
+
+#ifndef MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+#define MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+
+#include <dmlc/logging.h>
+#include "./mkldnn_cppwrapper.h"
+namespace mxnet {
+#if MXNET_USE_MKLDNN == 1
+template <typename DType>
+struct MKLMemoryDescriptorBase {
```

Comment:
after nnvm you can try to do this in passes. i.e. insert layout swap layer before and after all mkldnn layers